hr_employee_language/models/hr_employee_language.py

```diff
@@ -1,19 +1,44 @@
 # Copyright (C) 2017-Today: Odoo Community Association (OCA)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import fields, models, tools
+from odoo import _, fields, models
 
 
 class HrEmployeeLanguage(models.Model):
     _name = "hr.employee.language"
     _description = "HR Employee Language"
 
-    name = fields.Selection(
-        tools.misc.scan_languages(), string="Language", required=True
-    )
+    def _get_selection(self):
+        """
+        generate language selection from available on db
+        """
+        selection = []
+        for lang in self.env["res.lang"].search([("active", "in", (True, False))]):
+            selection += [
+                (_("%(code)s", code=lang.code), _("%(name)s", name=lang.name))
+            ]
+        return selection
+
+    name = fields.Selection(_get_selection, string="Language", required=True)
     description = fields.Char()
     employee_id = fields.Many2one("hr.employee", string="Employee", required=True)
     can_read = fields.Boolean(string="Read", default=True)
     can_write = fields.Boolean(string="Write", default=True)
     can_speak = fields.Boolean(string="Speak", default=True)
     can_listen = fields.Boolean(string="Listen", default=True)
+
+    def _compute_display_name(self):
+        """
+        redefine record name to show languages name instead languages code
+        """
+        result = []
+        all_lang = self.env["res.lang"].search([("active", "in", (True, False))])
+        self_data = self if self else all_lang
+        for record in self_data:
```

Comment:
I don't understand the point for this code.
What can't you just loop computed records as usual?
```suggestion
        for record in self:
```

Reply:
I did so because if I use your suggestion, that was my first option, test fails with:
 ```
File "/__w/hr/hr/hr_employee_language/tests/test_add_lang.py", line 33, in test_compute_display_name
    langs_name[0][1],
IndexError: list index out of range
```
Maybe the test is not well done.